src/main/java/com/google/sps/data/ChromeOSDevice.java

```diff
@@ -0,0 +1,110 @@
+package com.google.sps.data;
+
+/*
+ * Class representing a Chrome OS Device.
+ * See https://developers.google.com/admin-sdk/directory/v1/reference/chromeosdevices
+ * This class is an incomplete list of all chrome os device properties, instead containing
+ * only the subset of properties related to the webapp.  These properties can be expanded
+ * if the scope changes.
+ * ChromeOSDevice should not be mutated.
+ */
+public class ChromeOSDevice {
+
+  private String annotatedAssetId;
+  private String annotatedLocation;
+  private String annotatedUser;
+  private final String deviceId;
+  private final String serialNumber;
+
+  public ChromeOSDevice(
+      String annotatedAssetId,
+      String annotatedLocation,
+      String annotatedUser,
+      String deviceId,
+      String serialNumber) {
+    this.annotatedAssetId = annotatedAssetId;
+    this.annotatedLocation = annotatedLocation;
+    this.annotatedUser = annotatedUser;
+    this.deviceId = deviceId;
+    this.serialNumber = serialNumber;
+    sanitize();
+  }
+
+  // The Gson deserializer doesn't actually call the object's constructor, so we defined a custom
+  // deserializer which calls the sanitize() method. To keep everything consistent, the
+  // ChromeOSDevice constructor also calls the sanitize() method so the same standard is held
+  public void sanitize() {
+    annotatedAssetId = (annotatedAssetId == null) ? "" : annotatedAssetId;
+    annotatedLocation = (annotatedLocation == null) ? "" : annotatedLocation;
+    annotatedUser = (annotatedUser == null) ? "" : annotatedUser;
+  }
+
+  public ChromeOSDevice copy() {
+    return new ChromeOSDevice(getAnnotatedAssetId(),
+                              getAnnotatedLocation(),
+                              getAnnotatedUser(),
+                              getDeviceId(),
+                              getSerialNumber());
+  }
+
+  public String getAnnotatedField(String fieldName) throws IllegalArgumentException {
```

Comment:
suggestion: define a enum for the field name, e.g.
```
public enum AnnotatedField {
    ASSET_ID,
    LOCATION,
    USER
}
```

And then use `AnnotatedField` as the param of this function.

Reply:
I added in an enum for Annotated Fields, but rather than using it as a param for the `getAnnotatedField` function, I partially moved some of the logic into the enum by associating each option directly with a getter function, `getField()`.  So now you can create an annotated field e.g. `AnnotatedField location = AnnotatedField.LOCATION` and then call `location.getField(device)` to get the device's annotatedLocation.